pkg/util/probe.go

```diff
@@ -0,0 +1,111 @@
+package util
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+/*
+Common functionality to add readiness/liveness Probe to a container using:
+
+* default values:
+readinessProbe 	:= common.Probe{ProbeType: "readiness"}
+livenessProbe 	:= common.Probe{ProbeType: "liveness"}
+
+* custom values:
+readinessProbe := common.Probe{
+	ProbeType: 		"readiness",
+	Command:		"/to/some/command",
+	InitialDelaySeconds: 	20,
+ }
+livenessProbe := common.Probe{ProbeType: "liveness"}
+
+In container definition
+Containers: []corev1.Container{
+	{
+		Name:  "nova-conductor",
+		ReadinessProbe: readinessProbe.GetProbe(),
+		LivenessProbe:  livenessProbe.GetProbe(),
+		...
+*/
+
+const (
+	// default Readiness values
+	defaultReadinessInitialDelaySeconds int32 = 5
+	defaultReadinessPeriodSeconds       int32 = 15
+	defaultReadinessTimeoutSeconds      int32 = 3
+	defaultReadinessFailureThreshold    int32 = 3
+	// default Liveness values
+	defaultLivenessInitialDelaySeconds int32  = 30
+	defaultLivenessPeriodSeconds       int32  = 60
+	defaultLivenessTimeoutSeconds      int32  = 3
+	defaultLivenessFailureThreshold    int32  = 5
+	defaultCommand                     string = "/openstack/healthcheck"
+)
+
+const (
+	readiness string = "readiness"
+	liveness  string = "liveness"
+)
+
+type probeType string
+
+// Probe details
+type Probe struct {
+	// ProbeType, either readiness, or liveness
+	ProbeType           probeType
+	Command             string
+	InitialDelaySeconds int32 // min value 1
+	PeriodSeconds       int32 // min value 1
+	TimeoutSeconds      int32 // min value 1
+	FailureThreshold    int32 // min value 1
+}
+
+// GetProbe -
+func (p *Probe) GetProbe() *corev1.Probe {
```

Comment:
If you changed the signature to:

`func (p *Probe) ApplyProbe(probe *corev1.Probe) {`

and the modified the passed-in probe rather than creating a new one, you would also be able to use this util function in apply methods such as CreateOrUpdate()

Reply:
yes that sounds good. I'll add an ApplyProbe this in a follow up.